app/controllers/predicates/TaxYearAction.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.predicates
+
+import common.SessionValues
+import config.AppConfig
+import models.User
+import play.api.Logger
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.Results._
+import play.api.mvc._
+import javax.inject.Inject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TaxYearAction @Inject()(taxYear: Int)(
+  appConfig: AppConfig,
+  val messages: MessagesApi
+) extends ActionRefiner[User, User] with I18nSupport {
+
+  implicit val executionContext: ExecutionContext = ExecutionContext.global
+  lazy val logger: Logger = Logger.apply(this.getClass)
```

Comment:
nothing is being logged no longer needed

Reply:
we should copy the log that was there before